components/hybrid_engine/src/observer/snapshot.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{KvEngine, SnapshotContext};
+use raftstore::coprocessor::{
+    dispatcher::BoxSnapshotObserver, CoprocessorHost, ObservedSnapshot, SnapshotObserver,
+};
+use range_cache_memory_engine::{RangeCacheMemoryEngine, RangeCacheSnapshot};
+
+use crate::new_in_memory_snapshot;
+
+/// RangeCacheSnapshotPin pins data of a RangeCacheMemoryEngine during taking
+/// snapshot. It prevents the data from being evicted or deleted from the cache.
+pub struct RangeCacheSnapshotPin {
```

Comment:
As discussed offline, this SnapshotObserver is useless and should be removed. If you don't want to do it, please at least add a `TODO`.

Reply:
Added, and will open an issue for tracking the TODO.

Reply:
Here is the issue https://github.com/tikv/tikv/issues/17492